docker-build/Dockerfile

```diff
@@ -1,16 +1,10 @@
-FROM alpine:3.6
-
-LABEL description="Docker image for caddy+forwardproxy plugin."
-LABEL maintainer="SergeyFrolov@colorado.edu"
-
-RUN apk add --no-cache ca-certificates bash curl
-
-RUN curl --fail https://getcaddy.com | bash -s http.forwardproxy
-
-COPY gen_caddyfile_and_start.sh /bin/
-
-VOLUME /root/.caddy
-
-EXPOSE 80 443 2015
-
-ENTRYPOINT /bin/gen_caddyfile_and_start.sh
+FROM golang:1.14.15 AS builder
+RUN git clone https://github.com/caddyserver/forwardproxy.git
+WORKDIR /go/forwardproxy/cmd/caddy 
+RUN go build caddy.go
+
+FROM ubuntu:20.04
```

Comment:
Why ubuntu? Why not `FROM scratch`, isn't the point of using a builder image is to reduce overall image size by shipping only what caddy needs to run?

Reply:
Yes, however, "Scratch" image gives you an empty file system, That is all it does. scratch on it's own does absolutely nothing, and has nothing in it.